### PR TITLE
Github: CI/CD for gateway deployment to staging.

### PR DESCRIPTION
**Pull request by sk-enya**
:clipboard: Add associated issues, tickets, docs URL here.

closes: https://github.com/bobanetwork/boba-aws-infrastructure/issues/35

## Overview

Added pipeline to build & deploy the gateway to staging s3 bucket.

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Created workflow with required steps and env configuration.

## Testing

Tested on the separate repo.

### CODE REVIEW

File: .github/workflows/gateway-deploy-stage.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Print current tag
+        run: echo ${{ steps.tag.outputs.VERSION }}
+      - name: Build app stage
+        working-directory: ./packages/boba/gateway
+        run: yarn build:prod
+        env:
+          SKIP_PREFLIGHT_CHECK: true
+          REACT_APP_POLL_INTERVAL: 15000
+          REACT_APP_WALLET_VERSION: 0.6.1
+          REACT_APP_STATUS: 'maintainance'
```
Comment (boyuan-chen) [inline .github/workflows/gateway-deploy-stage.yml]: This should not be maintenance?

**Review by boyuan-chen — APPROVED** [review]

**Review by InoMurko — APPROVED** [review]

**Comment by sk-enya** [comment]
Waiting for secrets to be get added into the the github secrets mangers! so please don't merge.

**Comment by InoMurko** [comment]
status?

**Comment by sk-enya** [comment]
will merge and validate the changes today.

**Pull request merged**